engine/src/test/scala/cromwell/engine/workflow/WorkflowDockerLookupActorSpec.scala

```diff
@@ -110,6 +117,36 @@ class WorkflowDockerLookupActorSpec extends TestKitSuite("WorkflowDockerLookupAc
     hashResponses should equal(Set(LatestSuccessResponse, OlderSuccessResponse))
   }
 
+  // BA-6495
+  it should "not fail and enter terminal state when response for certain image id from DockerHashingActor arrived after the self-imposed timeout" in {
+    val lookupActor = TestFSMRef(new WorkflowDockerLookupActor(workflowId, dockerHashingActor.ref, isRestart = false, EngineServicesStore.engineDatabaseInterface))
+
+    lookupActor ! LatestRequest
+
+    val timeout = DockerHashActorTimeout(LatestRequest)
+
+    // The WorkflowDockerLookupActor should not have the hash for this tag yet and will need to query the dockerHashingActor.
+    dockerHashingActor.expectMsg(LatestRequest)
+    // WorkflowDockerLookupActor actually sends DockerHashActorTimeout to itself
+    lookupActor.tell(timeout, lookupActor)
+
+    val failedRequest: WorkflowDockerLookupFailure = receiveOne(2 seconds).asInstanceOf[WorkflowDockerLookupFailure]
+    failedRequest.request shouldBe LatestRequest
+
+    lookupActor ! LatestRequest
+    dockerHashingActor.expectMsg(LatestRequest)
+    dockerHashingActor.reply(LatestSuccessResponse) // responding for previously timeouted request
+    dockerHashingActor.reply(LatestSuccessResponse) // responding for current request
+
+    val hashResponse = receiveOne(2 seconds)
+    hashResponse shouldBe LatestSuccessResponse
+
+    // Give WorkflowDockerLookupActor a chance to finish its unfinished business
+    Thread.sleep(5000)
```

Comment:
can this be an `eventually { ... }`?

Reply:
The problem is that in bad scenario, state will be 'Running` for certain amount of time, and then change to `Terminal`. In this case `eventually` will mark tests as succeeded if it checks the state before it changed.